content/en/docs/collector/build-connector.md

```diff
@@ -0,0 +1,544 @@
+    - config.go - A file to define the connector's settings
+    - factory.go - A file to create instances of the connector
+
+### Create your connector settings in config.go
+
+In order to be instantiated and participate in pipelines, the Collector needs to
+identify your connector and properly load its settings from within its
+configuration file.
+
+In order to be able to give your connector access to its settings, create a
+Config struct. The struct must have an exported field for each of the
+connector’s settings. The parameter fields added will be accessible from the
+config.yaml file. Their name in the configuration file is set through a struct
+tag. Create struct and add parameters. You can optionally add a validator
+function to check if the given default values are valid for an instance of your
+connector.
+
+    package exampleconnector
+
+
+    import "fmt"
+
+
+    type Config struct {
+        AttributeName string `mapstructure:"attribute_name"`
+    }
+
+
+    func (c *Config) Validate() error {
+        if c.AttributeName == "" {
+    	    return fmt.Errorf("attribute_name must not be empty")
+        }
+        return nil
+    }
+
+Further details about mapstructure can be found at
+[Go mapstructure](https://pkg.go.dev/github.com/mitchellh/mapstructure).
+
+## Implement the Factory
+
+To instantiate the object, you will need to use the `NewFactory` function
+associated with each of the components. We will use the `connector.NewFactory`
+function. The `connector.NewFactory` function instantiates and returns a
+`connector.Factory` and it requires the following parameters:
+
+- `component.Type`: a unique string identifier for your connector across all
+  Collector’s components of the same type. This string also acts as the name to
+  refer to the connector by.
+- `component.CreateDefaultConfigFunc`: a reference to a function that returns
+  the default component.Config instance for your connector.
+- `...FactoryOption`: the slice of `connector.FactoryOptions` will determine
+  what type of signal your connector is capable of processing.
+
+1.  Create factory.go file and define the unique string to identify your
+    connector as a global constant.
+
+         const (
+           defaultVal = "request.n"
+           // this is the name used to refer to the connector in the config.yaml
+           typeStr = "example"
+         )
+
+2.  Create the default configuration function. This is how you choose to
+    initialize your connector object with default values.
+
+         func createDefaultConfig() component.Config {
+            return &Config{
+                AttributeName: defaultVal,
+            }
+         }
+
+3.  Define the type of the connector you will work with(this is a
+    factoryOption). A connector connects two pipelines, normally two differently
+    typed pipelines. We have to define the type of the exported end of the
+    connector and the receiver end of the connector.
```

Comment:
nit: trying to clarify wording a bit
```suggestion
3.  Define the type of the connector you will work with. This will be passed as a
    factory option. Usually, a connector connects two pipelines of different types.
    We have to define the type of the exported end of the
    connector and the receiver end of the connector.
```